docs/code-generation.md

```diff
@@ -0,0 +1,135 @@
+# Code generation
+
+In order to keep the code for all the service controllers consistent, we will
+use a strategy of generating the custom resource definitions and controller
+code stubs for new AWS services.
+
+To generate the CRDs and controller stub code, we investigated a number of
+options:
+
+* home-grown custom code generator
+* [kudo](https://kudo.dev)
+* [kubebuilder](github.com/kubernetes-sigs/kubebuilder)
+* a hybrid custom code generator + sigs.kubernetes.io/controller-tools (CR)
+
+The original AWS Service Operator used a [custom-built generator][1] that
+processed [YAML files][2] describing the AWS service and used [templates][3] to
+[generate CRDs][4], the [controller code][5] itself and the [Go types][6] that
+mirror the CRDs. It's worth noting that the CRDs *and* the controller code that
```

Comment:
s/mirror the CRDs/represent the CRDs in memory

Reply:
done.